backend/app/project/routes.py

```diff
@@ -3,6 +3,14 @@
 from app.models import Project
 from . import project
 
+
+# get all endpoints
+@project.route("/projects", methods=["GET"])
```

Comment:
@peterghrong minor issue, do we not want to leave it as just ""?

Reply:
^^

Reply:
thanks @CurtisMIT

Reply:
wait but also this is the same end point as here - https://github.com/hack4impact-mcgill/mu-crm-tool/blob/90cc6a54cb735e8ce85a0d428aea9e1386630530/backend/app/project/routes.py#L8 ??

Reply:
I had ` @project.route("", methods=["GET"]) ` initially but now since "get_projects_of_type():" exists which it goes through the same route `@project.route("", methods=["GET"])`, it was not returning the right things. 

I also thought that it would be preferable to keep the get_all routes consistent.
eg. `@project.route("/types", methods=["GET"])`

Regarding the other pr, I pulled this branch into the other branch for some reason, it was an amateur mistake and I should have just undid the git pull before I opened that pr. I'm really sorry for all this confusion. 

Lemme know how you still think if it's better to  change it to `@project.route("", methods=["GET"]) `, I'll get it fixed ASAP

Reply:
@peterghrong @celinehuang Yea in hindsight I maybe should have seen this coming, thanks for pointing out the clash again, mb missed it again. When we're sorting/filtering for more than one parameter, would we want to do it in the style below instead? As idk if we want to create a separate route for every filter we want to apply for projects. I've rewritten it so that we can freely add future request arguments if needed. It compiles and passes the tests. Any thoughts?

```
# get projects according to specified arguments (if any)
@project.route("", methods=["GET"])
def get_projects():
    projects = Project.query.all()

    type = request.args.get("type")
    if type is not None:
        if type == "":
            abort(404, "Project type invalid")
        else:
            projects = list(filter(lambda project: (project.type == type), projects))
    
    # add addition request arguments below

    return jsonify(Project.serialize_list(projects))
```

Inspired by these links:

scroll to (allow filtering, sorting, and pagination)
- https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/

scroll to (Using Query Arguments)
- https://www.digitalocean.com/community/tutorials/processing-incoming-request-data-in-flask

Reply:
hmm it's actually good to have the return all project type 'helper' endpoint because in frontend we have to render all possible filters anyways? Although now that I look at it, should we add a query parameter in the endpoint..?

Reply:
Yea I agree with the `"/types"` endpoint. The code segment I pasted is addressing the `""` endpoint, so that any filtering of projects can be done through that end point. To resolve the conflict for the `""` route.

Reply:
Thanks Curtis for bringing up the more proper conventions we should follow when creating endpoints. @peterghrong the snippet Curtis posted is a lot cleaner than the way I had originally suggested so I think we can remove this endpoint https://github.com/hack4impact-mcgill/mu-crm-tool/blob/28bdaf8681c291d7e46fb1ea01216bb2c64716f3/backend/app/project/routes.py#L16  that's in main and replace it with what Curtis had suggested